sycl/include/sycl/ext/oneapi/experimental/root_group.hpp

```diff
@@ -78,6 +85,75 @@ template <int Dimensions> class root_group {
 
   bool leader() const { return get_local_id() == 0; };
 
+  template <execution_scope Scope, typename RetTy>
+  using checkScopeTy = std::enable_if_t<(Scope == execution_scope::work_item ||
```

Comment:
Can this go into protected/private?

Reply:
Moved [0b2de74](https://github.com/intel/llvm/pull/12653/commits/0b2de7458bd7fa236bf653d59b855ccd28edbf1f)